src/Hazelcast.Net.Linq.Async/ObjectReader.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+// http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Hazelcast.Core;
+using Hazelcast.Sql;
+
+namespace Hazelcast.Linq
+{
+    // Implements IAsyncEnumerator and reads result.
+    internal class ObjectReader<TResult> : IAsyncEnumerator<TResult>
+    {
+        private readonly QueryProvider _queryProvider;
+        private readonly Expression _expression;
+        private IAsyncEnumerator<SqlRow>? _sqlEnumerator;
+        private ISqlQueryResult? _queryResult;
+        private Func<SqlRow, TResult>? _projector;
+        private readonly CancellationToken _cancellationToken;
+
+#pragma warning disable 8618 Non-nullable variable must contain a non-null value when exiting constructor. Consider declaring it as nullable.
```

Comment:
Mixed feeling about the warning. Cannot we fix the original issue?

Reply:
Warning is  for `Current` which is not initialized at ctor but `ObjectReader` is used as enumerator. So, that wouldn't be problem as long as result of `MoveNextAsync()` is obeyed.

Reply:
`Current = null!`

Reply:
Fixed at https://github.com/hazelcast/hazelcast-csharp-client/pull/752/commits/4506e1ba5fd0bae9f9c763cf789a39e0f652006f